lib/ansible/modules/cloud/azure/azure_rm_trafficmanager_facts.py

```diff
@@ -0,0 +1,372 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Hai Cao, <t-haicao@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: azure_rm_trafficmanager_facts
+
+version_added: "2.7"
+
+short_description: Get Traffic Manager profile facts
+
+description:
+    - Get facts for a specific Traffic Manager profile or all Traffic Manager profiles.
+
+options:
+    name:
+        description:
+            - Limit results to a specific Traffic Manager profile.
+    resource_group:
+        description:
+            - The resource group to search for the desired Traffic Manager profile
+    tags:
+        description:
+            - Limit results by providing a list of tags. Format tags as 'key' or 'key:value'.
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Hai Cao <t-haicao@microsoft.com>"
+'''
+
+EXAMPLES = '''
+    - name: Get facts for one Traffic Manager profile
+      azure_rm_trafficmanager_facts:
+        name: Testing
+        resource_group: TestRG
+
+    - name: Get facts for all Traffic Manager profiles
+      azure_rm_trafficmanager_facts:
+
+    - name: Get facts by tags
+      azure_rm_trafficmanager_facts:
+        tags:
+          - Environment:Test
```

Comment:
We should also support tags being a dict where the key is the tag name and the value is the value, e.g.

```
tags:
  Environment: test
```

Reply:
I checked several existing Azure facts modules and it seems that support tags being a list has been a convention. One example: https://github.com/ansible/ansible/blob/4c8808ec9df095a0bf5d9252eb65a07063843bcc/lib/ansible/modules/cloud/azure/azure_rm_resourcegroup_facts.py#L53-L57

Reply:
Ok food for thought, I know the `tags` option for non facts modules can take in the dict format (or maybe the reverse). This should be something we try and unify in the future.